modules/zenith-public/lua/4-additive/items/exclusive_equipment.lua

```diff
@@ -0,0 +1,78 @@
+-----------------------------------
+-- Exclusive Equipment Groups
+-- Prevents dual wielding of items within the same exclusion group
+-----------------------------------
+require('modules/module_utils')
+-----------------------------------
+
+local m = Module:new('exclusive_equipment')
+
+-- Define exclusive groups - items in the same group cannot be equipped together
+-- To add new groups, create a new entry with the group name as key and a table
+-- of item IDs mapped to their internal names as the value
+local exclusiveGroups =
+{
+    -- Multi-attack clubs that cannot be dual wielded
+    multiAttackClubs =
+    {
+        [xi.item.KRAKEN_CLUB] = 'kraken_club',
+        [xi.item.OCTAVE_CLUB] = 'octave_club',
+    },
+    -- Add more groups here as needed
+    -- Example:
+    -- specialSwords =
+    -- {
+    --     [xi.item.SOME_SWORD] = 'some_sword',
+    --     [xi.item.OTHER_SWORD] = 'other_sword',
+    -- },
+}
+
+-- Helper function to check if an item belongs to an exclusive group
+-- @param itemId number: The item ID to check
+-- @return string|nil groupName: The name of the exclusive group, or nil if not found
+-- @return table|nil items: The items table for the group, or nil if not found
+local function getExclusiveGroup(itemId)
+    for groupName, items in pairs(exclusiveGroups) do
+        if items[itemId] then
+            return groupName, items
+        end
+    end
+
+    return nil, nil
+end
+
+-- Helper function to handle equipment conflicts
+-- @param target object: The player object
+local function handleEquipmentConflict(target)
+    target:timer(50, function(targetArg)
+        -- Unequip both MAIN and SUB weapons when conflict detected
+        targetArg:unequipItem(xi.slot.MAIN)
+        targetArg:unequipItem(xi.slot.SUB)
+        targetArg:messageBasic(xi.msg.basic.ITEM_UNABLE_TO_USE)
+    end)
+end
+
+-- Apply overrides for all exclusive items
+for groupName, groupItems in pairs(exclusiveGroups) do
+    for itemId, itemName in pairs(groupItems) do
+        local itemPath = fmt('xi.items.{}', itemName)
+        xi.module.ensureTable(itemPath)
+
+        m:addOverride(itemPath .. '.onItemEquip', function(player, item)
+            -- Get currently equipped items
+            local mainWeapon = player:getEquipID(xi.slot.MAIN)
+            local subWeapon = player:getEquipID(xi.slot.SUB)
```

Comment:
```suggestion
            super(player, item)

            -- Get currently equipped items
            local mainWeapon = player:getEquipID(xi.slot.MAIN)
            local subWeapon = player:getEquipID(xi.slot.SUB)
```

forgot to check for this on my first review pass